EventFilter/L1TRawToDigi/plugins/implementations_stage2/GlobalAlgBlkUnpacker.cc

```diff
@@ -40,6 +40,12 @@ namespace l1t {
       // Loop over multiple BX and then number of EG cands filling collection
       int numBX = 0;  //positive int to count BX
       for (int bx = firstBX; bx <= lastBX; bx++) {
+        // Check to make sure that we are receiving the first expected header here
+        // If we are not, throw a more informative error than the failure of the following "at->" call
+        if (res_->size(bx) == 0 and block.header().getID() != initialBlkID)
+          throw(
+              std::runtime_error("The GlobalAlgBlk unpacker result vector is size 0, but is not receiving the first "
+                                 "expected header ID! This may be due to corrupted, or poorly formatted events."));
```

Comment:
- I was expecting the check to be after the `if` statement, e.g. [*] (if the `if` condition is true, an element is added and there is no need to check, the vector won't be empty for that bx).
 - why not use the standard `cms::Exception` ?
 - why not add more specific info in the exception message, e.g. values of `bx`, `block.header().getID()`, `uGTBoard`, etc ? (on the specifics of the error message, I defer to L1T)

[*]
```diff
        // If this is the first block on first board, instantiate GlobalAlg so it is there to fill from mult. blocks
        if (block.header().getID() == initialBlkID && uGTBoard == 0) {
          LogDebug("L1T") << "Creating GT Algorithm Block for BX =" << bx << std::endl;
          GlobalAlgBlk talg = GlobalAlgBlk();
          res_->push_back(bx, talg);
        }
+       // throw exception if res_ is empty for this BX (indication of corrupted RAW data)
+       else if (res_->isEmpty(bx)) {
+         throw cms::Exception("InvalidGlobalAlgBlkBxCollection") << "[the message]"
+       }
```

Reply:
This is better. I'll make those changes.